sdk/storage/MigrationGuide.md

```diff
@@ -0,0 +1,479 @@
+# Migration Guide: From Azure Storage Legacy CPP SDK to Azure Stroage CPP SDK
+
+This guide intends to assist customers in migrating from legacy versions of the Azure Storage C++ library for Blobs to version 12.
+While this guide is generally applicable to older versions of the SDK, it was written with v7.5 in mind as the starting point.
+It will focus on side-by-side comparisons for similar operations between the [v7.5 package](https://github.com/Azure/azure-storage-cpp) and [v12 package](https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/storage).
+Familiarity with the legacy client library is assumed. For those new to the Azure Storage Blobs client library for C++, please refer to the [Quickstart](https://docs.microsoft.com/azure/storage/blobs/quickstart-blobs-c-plus-plus) for the v12 library rather than this guide.
+
+## Table of contents
+
+- [Migration benefits](#migration-benefits)
+- [General changes](#general-changes)
+  - [Package and namespaces](#package-and-namespaces)
+  - [Authentication](#authentication)
+  - [Client structure](#client-structure)
+- [Migration samples](#migration-samples)
+  - [Creating a Container](#creating-a-container)
+  - [Uploading Blobs to a Container](#uploading-blobs-to-a-container)
+  - [Downloading Blobs from a Container](#downloading-blobs-from-a-container)
+  - [Listing Blobs in a Container](#listing-blobs-in-a-container)
+  - [Managing Blob Metadata](#managing-blob-metadata)
+  - [Content Hashes](#content-hashes)
+  - [Resiliency](#resiliency)
+- [Additional information](#additional-information)
+
+## Migration benefits
+
+To understand why we created our version 12 client libraries, you may refer to the Tech Community blog post, [Announcing the Azure Storage v12 Client Libraries](https://techcommunity.microsoft.com/t5/azure-storage/announcing-the-azure-storage-v12-client-libraries/ba-p/1482394) or refer to our video [Introducing the New Azure SDKs](https://aka.ms/azsdk/intro).
+
+Included are the following:
+- Thread-safe synchronous APIs
+- More comprehensive feature parity
+- Dropped dependency of Boost and cpprestsdk
+- Improved performance
+- Consistent and idiomatic code organization, naming, and API structure, aligned with a set of common guidelines
+- The learning curve associated with the libraries was reduced
+
+Note: The blog post linked above announces deprecation for previous versions of the library.
+
+## General changes
+
+### Package and namespaces
+
+Package names and the namespaces root for version 12 Azure client libraries follow the pattern `Azure::[Area]::[Service]` where the legacy libraries followed the pattern `azure::[area]::[service]`.
+
+In this case, to install the legacy v7.5 package with vcpkg:
+```bash
+vcpkg install azure-storage-cpp
+```
+
+It is now the following for v12:
+```bash
+vcpkg install azure-storage-blobs-cpp
+```
```

Comment:
```suggestion

Version 12 packages are installed with:
```bash
vcpkg install azure-storage-blobs-cpp
```
```
Previously the v7.5 package was installed with:
```bash
vcpkg install azure-storage-cpp
```